lib/ansible/modules/cloud/vmware/vmware_host_snmp.py

```diff
@@ -0,0 +1,431 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2018, Christian Kotte <christian.kotte@gmx.de>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+module: vmware_host_snmp
+short_description: Configures SNMP on an ESXi host system
+description:
+- This module can be used to configure the embedded SNMP agent on an ESXi host.
+version_added: 2.8
+author:
+- Christian Kotte (@ckotte)
+notes:
+- Tested on vSphere 6.5
+- You need to reset the agent (to factory defaults) if you want to delete all community strings or trap targets
+requirements:
+- python >= 2.6
+- PyVmomi
+options:
+  state:
+    description:
+        - Enable, disable, or reset the SNMP agent.
+    type: str
+    choices: [ enabled, disabled, reset ]
+    default: 'disabled'
+  community:
+    description:
+        - List of SNMP community strings.
+    type: list
+  snmp_port:
+    description:
+        - Port used by the SNMP agent.
+    type: int
+    default: 161
+  trap_targets:
+    description:
+        - A list of trap targets.
+        - You need to use C(hostname), C(port), and C(community) for each trap target.
+    default: []
+  trap_filter:
+    description:
+        - Comma separated list of trap oids for traps not to be sent by agent.
+        - E.g. 1.3.6.1.4.1.6876.4.1.1.0,1.3.6.1.4.1.6876.4.1.1.1
+        - Use value C(reset) to clear settings.
+    type: str
+  send_trap:
+    description:
+        - Send a test trap to validate the configuration.
+    type: bool
+    default: false
+  hw_source:
+    description:
+        - Source hardware events from IPMI sensors or CIM Indications.
+        - The embedded SNMP agent receives hardware events either from IPMI sensors C(sensors) or CIM indications C(indications).
+    type: str
+    choices: [ indications, sensors ]
+    default: 'indications'
+extends_documentation_fragment: vmware.documentation
+'''
+
+EXAMPLES = r'''
+- name: Enable and configure SNMP
+  vmware_host_ad_auth:
+    hostname: '{{ vcenter_hostname }}'
+    username: '{{ vcenter_username }}'
+    password: '{{ vcenter_password }}'
+    esxi_hostname: '{{ esxi_hostname }}'
+    domain: example.local
+    join_user: adjoin
```

Comment:
I think these parameters are not related to this module.

Reply:
fixed